### PR TITLE
Documenting selective override

### LINKED ISSUE #1226: Need documentation for Selective Override  (Falco PR# 2981)
/area documentation

**What would you like to be added**:
Need documentation for https://github.com/falcosecurity/falco/pull/2981

**Why is this needed**:

### PR DESCRIPTION
**Pull request by mikegcoleman**
**What type of PR is this?**

/kind cleanup

/kind content

Any specific area of the project related to this PR?**
/area documentation

**What this PR does / why we need it**:

Adds documentation for the new selective override feature

**Which issue(s) this PR fixes**:

Fixes #1226 

**Special notes for your reviewer**:

### CODE REVIEW

File: content/en/docs/rules/overriding.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+`replace` allows you to replace the value of an list, macro or macro key
+
+{{% alert color="warning" %}}
+`append` and `replace` cannot be used together. Trying to apply both `append` and `replace` to a key will result in an error. 
+{{% /alert %}}
+
+The keys that can be overridden vary by rules component and action being taken:
+
+* Lists (`append` or `replace`): `{"items"}`
+* Macros (`append or replace`): `{"condition"}`
```
Comment (Andreagit97) [inline content/en/docs/rules/overriding.md:31]:
```suggestion
* Macros (`append` or `replace`): `{"condition"}`
```

File: content/en/docs/rules/overriding.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    output: replace
+```
+The rule `program_accesses_file` would trigger when `ls`/`cat` either used `open` on a file, but not if the user was root.
+
+The new output message would be `A file (user=%user.name command=%proc.cmdline file=%fd.name) was opened by a monitored program`
+
+## Appending to existing rules using `append:true` (deprecated)
+
+{{% alert color="warning" %}}
+This method has been depreciated and will be removed in Falco 1.0.
```
Comment (Andreagit97) [inline content/en/docs/rules/overriding.md:146]:
```suggestion
This method has been depreciated and will be removed in Falco 1.0.0 
```

File: content/en/docs/rules/overriding.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+  condition: and not user.name=root
+  output: A file (user=%user.name command=%proc.cmdline file=%fd.name) was opened by a monitored program
+  override: 
+    condition: append
+    output: replace
+```
+The rule `program_accesses_file` would trigger when `ls`/`cat` either used `open` on a file, but not if the user was root.
+
+The new output message would be `A file (user=%user.name command=%proc.cmdline file=%fd.name) was opened by a monitored program`
+
```
Comment (Andreagit97) [inline content/en/docs/rules/overriding.md:142]: Just an idea...maybe we can add yet another example on the `enabled` case.
I mean, something like this:

```yaml
- rule: test_rule
  desc: test rule description
  condition: evt.type = close
  output: user=%user.name command=%proc.cmdline file=%fd.name
  priority: INFO
  enabled: false

- rule: test_rule
  enabled: true
```

is now deprecated and you should use instead:


```yaml
- rule: test_rule
  desc: test rule description
  condition: evt.type = close
  output: user=%user.name command=%proc.cmdline file=%fd.name
  priority: INFO
  enabled: false

- rule: test_rule
  enabled: true
  override:
    enabled: replace
```

In this way we show the new correct usage of `enabled` WDYT?

**Review by Andreagit97 — COMMENTED** [review]
Amazing work! Just few nits

File: content/en/docs/rules/overriding.md (hunk 3 above)
Comment (Andreagit97) [reply content/en/docs/rules/overriding.md:142]: and also in this example, we can highlight that this usage is deprecated! Falco will output a warning if you use it
```yaml
- rule: test_rule
  desc: test rule description
  condition: evt.type = close
  output: user=%user.name command=%proc.cmdline file=%fd.name
  priority: INFO
  enabled: false

- rule: test_rule
  enabled: true
```

Comment (mikegcoleman) [reply content/en/docs/rules/overriding.md:143]: I made all the changes you suggested @Andreagit97  Can your please review. Thanks!

File: content/en/docs/rules/overriding.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+{{% alert color="warning" %}}
+`append` and `replace` cannot be used together. Trying to apply both `append` and `replace` to a key will result in an error. 
+{{% /alert %}}
+
+The keys that can be overridden vary by rules component and action being taken:
+
+* Lists (`append` or `replace`): `{"items"}`
+* Macros (`append` or `replace`): `{"condition"}`
+* Rules (`append`): `{"condition", "output", "desc", "tags", "exceptions"}`
```
Comment (LucaGuerra) [inline content/en/docs/rules/overriding.md:32]:
```suggestion
* Rules (`append`): `condition`, `output`, `desc`, `tags`, `exceptions`
```

And similar changes to the above and below :)

File: content/en/docs/rules/overriding.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+* Rules (`replace`):  `{"condition", "output", "desc", "priority", "tags", "exceptions", "enabled", "warn_evttypes", "skip-if-unknown-filter"}`
+
+## Examples of using the `ovverride` section
```
Comment (LucaGuerra) [inline content/en/docs/rules/overriding.md:35]:
```suggestion
## Examples of using the `override` section
```

File: content/en/docs/rules/overriding.md [hunk 6]
```diff
@@ -0,0 +1,273 @@
+---
+title: Overriding Rules
+description: Overriding Falco rules
+linktitle: Overriding Rules
+weight: 50
+---
+## Overview
+There may be cases where you need to adjust the behavior of the Falco-supplied {{< glossary_tooltip text="list" term_id="lists" >}}, {{< glossary_tooltip text="macro" term_id="macros" >}}, and  {{< glossary_tooltip text="rule" term_id="rules" >}}.
+
+To enable this Falco allows you to define multiple rules files. The additional rules files can be used to add new lists, macros and rules or to override (modify) existing ones.
```
Comment (LucaGuerra) [inline content/en/docs/rules/overriding.md:10]: We could note that you can also override in the same file, as long as the override is specified after the definition.

File: content/en/docs/rules/overriding.md [hunk 7]
```diff
@@ -0,0 +1,273 @@
+---
+title: Overriding Rules
+description: Overriding Falco rules
+linktitle: Overriding Rules
+weight: 50
```
Comment (LucaGuerra) [inline content/en/docs/rules/overriding.md:5]: I would suggest adding an alias to the old `appending` page so links won't be broken

File: content/en/docs/rules/overriding.md (hunk 6 above)
Comment (mikegcoleman) [reply content/en/docs/rules/overriding.md]: Done!

**Review by Andreagit97 — COMMENTED** [review]
LGTM! Thank you!

File: content/en/docs/rules/overriding.md [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+description: Overriding Falco rules
+linktitle: Overriding Rules
+aliases: 
+- /docs/rules/appending
+weight: 50
+---
+## Overview
+There may be cases where you need to adjust the behavior of the Falco-supplied {{< glossary_tooltip text="list" term_id="lists" >}}, {{< glossary_tooltip text="macro" term_id="macros" >}}, and  {{< glossary_tooltip text="rule" term_id="rules" >}}.
+
+You can override (modify) rules in falco two different ways:
```
Comment (incertum) [inline content/en/docs/rules/overriding.md:12]: Falco (upper case throughout?)

File: content/en/docs/rules/overriding.md [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+aliases: 
+- /docs/rules/appending
+weight: 50
+---
+## Overview
+There may be cases where you need to adjust the behavior of the Falco-supplied {{< glossary_tooltip text="list" term_id="lists" >}}, {{< glossary_tooltip text="macro" term_id="macros" >}}, and  {{< glossary_tooltip text="rule" term_id="rules" >}}.
+
+You can override (modify) rules in falco two different ways: 
+
+1) Define multiple rules files. The additional rules files can be used to add new lists, macros and rules or to override existing ones.
```
Comment (incertum) [inline content/en/docs/rules/overriding.md:14]: It doesn't fit in here but maybe you have an idea @mikegcoleman how to give a more concrete example. Somehow stating that a typical use case is to fetch the upstream Falco rules hosted in the rules repo and then customize them using the override feature.

File: content/en/docs/rules/overriding.md [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+
+##### `/etc/falco/falco_rules.yaml`
+
+```yaml
+- macro: access_file
+  condition: evt.type=open
+
+- rule: program_accesses_file
+  desc: track whenever a set of programs opens a file
+  condition: proc.name in (cat, ls) and (access_file)
```
Comment (incertum) [inline content/en/docs/rules/overriding.md:107]:
```suggestion
  condition: (access_file) and proc.name in (cat, ls)
```

Follow the style guide

File: content/en/docs/rules/overriding.md [hunk 11]
```diff
@@ -0,0 +1,323 @@
...
+The rule `program_accesses_file` would trigger when `ls`/`cat` either used `open`/`openat` on a file.
+
+### Append and replace items in a rule
+
+##### `/etc/falco/falco_rules.yaml`
+
+```yaml
+- rule: program_accesses_file
+  desc: track whenever a set of programs opens a file
+  condition: proc.name in (cat, ls) and evt.type=open
```
Comment (incertum) [inline content/en/docs/rules/overriding.md:130]: same follow style guide

File: content/en/docs/rules/overriding.md [hunk 12]
```diff
@@ -0,0 +1,323 @@
...
+    condition: append
+    output: replace
+```
+The rule `program_accesses_file` would trigger when `ls`/`cat` either used `open` on a file, but not if the user was root.
+
+The new output message would be `A file (user=%user.name command=%proc.cmdline file=%fd.name) was opened by a monitored program`
+
+### Enabling a disabled rule
+
+Using `enabled: true` is deprecated, and should be avoided. Falco 0.37.0 and later will display a warning If `enabled: true` is used.
```
Comment (incertum) [inline content/en/docs/rules/overriding.md:151]: warning if

File: content/en/docs/rules/overriding.md [hunk 13]
```diff
@@ -0,0 +1,323 @@
...
+```yaml
+- rule: test_rule
+  desc: test rule description
+  condition: evt.type = close
+  output: user=%user.name command=%proc.cmdline file=%fd.name
+  priority: INFO
+  enabled: false
+```
+
+##### `/etc/falco/falco_rules.local.yaml`
```
Comment (incertum) [inline content/en/docs/rules/overriding.md:164]:
```suggestion
##### `/etc/falco/falco_rules.local.yaml` (wrong usage example)
```

File: content/en/docs/rules/overriding.md [hunk 14]
```diff
@@ -0,0 +1,323 @@
...
```
Comment (incertum) [inline content/en/docs/rules/overriding.md:184]:
```suggestion
##### `/etc/falco/falco_rules.local.yaml` (correct usage example)
```

File: content/en/docs/rules/overriding.md [hunk 15]
```diff
@@ -0,0 +1,323 @@
...
+- rule: test_rule
+  enabled: true
+  override:
+    enabled: replace
+```
+
+## Appending to existing rules using `append:true` (deprecated)
+
+{{% alert color="warning" %}}
+This method has been depreciated and will be removed in Falco 1.0.0
```
Comment (incertum) [inline content/en/docs/rules/overriding.md:196]:
```suggestion
This method has been deprecated and will be removed in Falco 1.0.0 
```

File: content/en/docs/rules/overriding.md (hunk 11 above)
Comment (mikegcoleman) [reply content/en/docs/rules/overriding.md:130]: fixed :)

File: content/en/docs/rules/overriding.md (hunk 8 above)
Comment (mikegcoleman) [reply content/en/docs/rules/overriding.md:12]: Fixed :)

File: content/en/docs/rules/overriding.md (hunk 12 above)
Comment (mikegcoleman) [reply content/en/docs/rules/overriding.md:151]: Fixed :)

**Review by mikegcoleman — COMMENTED** [review]
Addressed these comments

File: content/en/docs/rules/overriding.md [hunk 16]
```diff
@@ -0,0 +1,323 @@
+---
+title: Overriding Rules
+description: Overriding Falco rules
+linktitle: Overriding Rules
+aliases: 
+- /docs/rules/appending
```
Comment (Andreagit97) [inline content/en/docs/rules/overriding.md:6]:
```suggestion
- appending
```

**Review by leogr — APPROVED** [review]

**Pull request merged**